### DISCUSSION TITLE
Does this tool re-download deleted podcast episodes?

### DISCUSSION THREAD

**Discussion by WolfgangDpunkt** [dvehrs/podget#66] [Q&A] [answered]
I didn't see a "skip already downloaded, but since deleted files" option. But if you can set it to delete old episodes automatically, there must also be a way to prevent podget from download them again, right?
Before I try it myself, I'll just ask, possibly as a suggestion to add this info in the readme ;-)

For me, this is the most important feature, because I remove episodes after finished listening to it, and it would be fatal if they are then downloaded again.
Thank you very much!

**Comment by dvehrs** [comment, marked as answer]
**Short answer:**  Yes it does keep a log of previously downloaded files.

**Long answer:**

When you configure Podget, you define a directory (DIR_LIBRARY) to store the downloaded files in.  Those can be subdivided into folders or not but that is up to how you configure it.  Inside that directory there is also a hidden directory called .LOG, that contains two files:  "done" and "errors".   When Podget attempts to download a file, the URL for that file will be added to one of the two files.  Those that have a problem downloading are added to "errors".  Those that succeed are added to "done".  

For general use when Podget is run again, it will attempt to download any file in the "errors" file again and if it succeeds it removes it from "errors" then adds it to "done".  

When Podget runs, it downloads the feed and extracts the URLs to download from it.  Those URLs are then handled sequentially.  First they are checked against the "done" file, and if found they skip to the next URL.  If they are not found, the URL is downloaded.  

This means that for the vast majority of feeds, Podget only downloads new files.  However there are some feeds that create their download URLs a bit dynamically.  Because the URL is changing for those downloads, Podget will download the file again.  However in my experience that is only a tiny minority of feeds and for those that it does happen to it appears to be something that happens when they make big changes to their server software.  So it's fairly rare for them too.  

For a bit of historical perspective, Podget was first released in 2005 and at the time like many others I was using a dial-up modem.  Bandwidth was not something that could be freely abused so working in simple methods to reduce any repeated downloads was critical.   

**Different Twist on Same Issue:**

If you have a feed that is dynamically creating its download URLs so they are being repeatedly downloaded there is one other option to limit downloads.  That is the --recent (or -r) option.  This allows you to set a limit on the number of files downloaded from any feed.  If you set it to 5 then it will only download the file most recent files no matter how many are defined in the feed list.  In that case, both files that are downloaded and those that aren't are added to the "done" file to try to prevent future downloads of them.  

By default Podget assumes that feeds will list their items in a descending manner.  That is newest items first and oldest last.  This appears to be the most common.  However if you have a feed that puts oldest first and newest last then for that feed in your serverlist file you can add the option OPT_FEED_ORDER_ASCENDING.   As long as you the feed order correct then using the --recent option will keep extra downloads to a minimum.  

However keep in mind that depending on how many feeds you want to download and the options required, you may need to define more than one serverlist, podgetrc and cron job.

**Comment by WolfgangDpunkt** [comment]
Thank you very much for this detailed explanation. That sounds very plausible.
 I am glad to have found this very useful application.
